Source/Dafny/Resolver.cs

```diff
@@ -1999,7 +1999,7 @@ public class ModuleBindings {
           // saying is that the Method/Predicate does not take any type parameters over and beyond what the enclosing type (namely, the
           // iterator type) does.
           // --- here comes the constructor
-          var init = new Constructor(iter.tok, "_ctor", new List<TypeParameter>(), iter.Ins,
+          var init = new Constructor(iter.tok, "_ctor", false, new List<TypeParameter>(), iter.Ins,
```

Comment:
I agree. In fact, there are many passes through the AST (like the ghost-interest visitor) that run in one of the phases of the resolver. It would be nice to put each one of those into a separate file (and perhaps clean up the call sites a bit, too).